modin/experimental/xgboost/xgboost.py

```diff
@@ -0,0 +1,499 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import os
+import time
+import logging
+from threading import Thread
+from typing import Dict, Optional
+from multiprocessing import cpu_count
+
+import xgboost as xgb
+import numpy as np
+import pandas
+
+from modin.developer import unwrap_row_partitions
+
+if os.environ.get("MODIN_ENGINE", "Ray").title() == "Ray":
+    import ray
+    from ray.services import get_node_ip_address
+else:
+    raise ValueError("Current version supports only Ray engine as MODIN_ENGINE.")
+
+LOGGER = logging.getLogger("[xgboost.modin]")
+
+
+def _start_rabit_tracker(num_workers: int):
+    """Start Rabit tracker. The workers connect to this tracker to share
+    their results."""
+    host = get_node_ip_address()
+
+    env = {"DMLC_NUM_WORKER": num_workers}
+    rabit_tracker = xgb.RabitTracker(hostIP=host, nslave=num_workers)
+
+    # Get tracker Host + IP
+    env.update(rabit_tracker.slave_envs())
+    rabit_tracker.start(num_workers)
+
+    # Wait until context completion
+    thread = Thread(target=rabit_tracker.join)
+    thread.daemon = True
+    thread.start()
+
+    return env
+
+
+class RabitContext:
+    """Context to connect a worker to a rabit tracker"""
+
+    def __init__(self, actor_id, args):
+        self.args = args
+        self.args.append(("DMLC_TASK_ID=[xgboost.modin]:" + actor_id).encode())
+
+    def __enter__(self):
+        xgb.rabit.init(self.args)
+        LOGGER.info("-------------- rabit started ------------------")
+
+    def __exit__(self, *args):
+        xgb.rabit.finalize()
+        LOGGER.info("-------------- rabit finished ------------------")
+
+
+@ray.remote
```

Comment:
shouldn't this be in separate module, like `xgboost_ray.py`?

Reply:
I think it will make sense in case supporting another distributed engines by modin.xgboost.

Reply:
Ray-specific functionality was moved to `xgboost_ray.py` file.